addons/website_sale/controllers/main.py

```diff
@@ -1227,3 +1232,61 @@ def get_combination_info_website(self, product_template_id, product_id, combinat
             })
         res['carousel'] = carousel_view
         return res
+
+    # ------------------------------------------------------
+    # Products Search Bar
+    # ------------------------------------------------------
+
+    @http.route('/shop/products/autocomplete', type='json', auth='public', website=True)
+    def products_autocomplete(self, term, limit, product_options, **kwargs):
+        """
+        Returns list of products according to the term and product options
+
+        Params:
+            term (str): search term written by the user
+            limit (int): number of products we want to return in the list
+            product_options (dict): product options set by the user (from snippet option)
+
+        Returns:
+            {} (dict): list of products
+        """
+        display_description = product_options['display_description']
+        display_price = product_options['display_price']
+        order_by = product_options['order_by']
+
+        domain = self._get_search_domain(term, None, None, display_description)
+        products = request.env['product.template'].search(domain, limit=limit, order=order_by)
+        pricelist = request.website.get_current_pricelist()
+
+        def get_product_info(product):
+            product_info = {
+                'id': product.id,
+                'name': product.name,
+                'url': product.website_url,
+            }
+
+            if display_description:
+                product_info['description'] = product.description_sale
+
+            if display_price:
+                first_possible_combination = product._get_first_possible_combination()
+                combination_info = product._get_combination_info(first_possible_combination,
+                                                                 add_qty=1,
+                                                                 pricelist=pricelist)
+                product_info.update({
+                    'price': combination_info['price'],
+                    'currency_symbol': pricelist.currency_id.symbol,
```

Comment:
in this case, adapt docstring, because instead to have a flat list of product, you will start with a sturctured dict

{
    'currency': 'usd',
    'pricelist': ...,
    'product_ids': {
           'name': '...',
           'price': '...',
           'description': '...',
           ...
     }
}

not sure about the real bytes won VS the readability/code